### ISSUE TITLE
Clicking Avatar for Local Sources Leads to "The source [x] isn't in your library" Error Message

### ISSUE THREAD

**Issue by professor-mesmerize** [ififfy/flipflip#250]
### Description
When clicking on the "avatar" for a local source (the little folder icon), a preview of the source doesn't play as expected. Instead, an error message is displayed, reading "The source [path of source] isn't in your library". This happens whenever I click on the local source avatar in any scenario - including in a scene and in the Source Library itself. Non-local sources (like bdsmlr) do preview correctly when clicking their avatar!

This bug first appeared for me in v3.1.0-beta1, but I assumed it was something wrong with my configuration for awhile and didn't report it because the sources themselves seem to be usable just fine when actually playing a scene! (also, weirdly, the number on the side of the source indicating the number of objects in it _does_ update when the source is played in a scene!) 

When it kept happening in v3.1.0-beta2, I investigated a bit more thoroughly, kept trying with a fresh install + a cleared `Application Support` folder, but the same issue kept occurring. Overall, I couldn't find any way to make it _not_ occur, so long as the source was local. But, good news - the sources themselves seem to be usable just fine when actually playing a scene! (also, weirdly, the number on the side of the source indicating the number of objects in it _does_ update when the source is played in a scene!)

### Deets
**Versions:**
- v3.1.0-beta2
- v3.1.0-beta1

**OS:**
- macOS 10.15.7

**Screenshots:**
_(From Library)_
<img width="862" alt="LibrarySourceBugScreenshot" src="https://user-images.githubusercontent.com/65079055/107120863-632be380-685d-11eb-8005-f455c23840b5.png">

_(From Scene)_
<img width="862" alt="SceneSourceBugScreenshot" src="https://user-images.githubusercontent.com/65079055/107120878-6a52f180-685d-11eb-96fa-fa99b850336a.png">

**Attachments:**
[data.json.txt](https://github.com/ififfy/flipflip/files/5936985/data.json.txt) `data.json` file (but in .txt format cause github won't accept .json files??)

### Repro Steps

1. Do a fresh install of FlipFlip by deleting not only `FlipFlip.app`, but also the supporting folder `~/Library/Application Support/flipflip`, and then reinstalling FlipFlip. (possibly optional! i had the same bug prior to doing a fresh install as well - i just also wanted to try a fresh install in case it was something with my cache or data.json file)
2. Add a source from a local directory (either directly into a scene or into your Source Library)
3. Click on the "Avatar" for the source

**Comment by StarrHelixx** [comment]
Hey! Thank you _so_ much for your detailed bug report! I was easily able to reproduce this (Mac only bug) and I've already pushed a fix (https://github.com/ififfy/flipflip/commit/848e50c6aa113533960dac8b043b9a85d9a3be9c)! 

Issue was with file path parsing and assumed windows file separator. I've fixed this in a few places, which I'm sure also wouldn't have worked properly. This is available now on `master` and will be included in the next release! (So happy you caught this before `v3.1.0` so ty again!)
